### DISCUSSION TITLE
Add an option to see hit counts in the console

### DISCUSSION THREAD

**Discussion by jonbell-lot23** [RecordReplay/devtools#7131] [Show and tell]
An idea from @markerikson -- it would be nice to have an option to see hits in the console to help you find the one you're looking for more easily. Especially since our console can be so small in its default configuration.

<img width="905" alt="image" src="https://user-images.githubusercontent.com/9154902/172978957-177bca44-4e65-43d0-9bb3-56a4a5c890cd.png">

Figma: https://www.figma.com/file/mAotNZRv6M5X0zHIFKPzFY/Replay-Design-Doc?node-id=12807%3A103428

**Comment by jasonLaster** [comment]
I think this is a tricky one because it is probably important to go back and think about the problems in order to land on the best solution. In the interest of time, here are some options

1. a tooltip on hover that shows offset and total
2. variables in the expression for $offset and $total which would be shown in autocomplete
3. show offset in the fastforward / rewind buttons all the time, if option enabled, on hover...
4. first explore message nesting which we may or may not have a discussion for, but then use the nesting as the place to show offset
5. decouple offset and total bacause most of the time it is offset you want to scan and total you want to reference. Offset is what is changing all the times

It'd be great to list out some of the problems we want to solve for / use cases we want to handle as well.

**Reply by markerikson** [reply]
I don't think any of these really match what I personally am looking for, tbh :)

In general, what I want to know while looking at these messages is:

- Where am I in the overall sequence for a given print statement?
- If I'm going to start using a given message to narrow down the focus range, how many are there?

Tooltips and expression variables don't help, because I want to see the numbers at a glance as I'm scrolling through.  Not sure "nesting" helps, because different print statements can be interleaved.

To be clear, the suggestion is that **these would be a toggleable setting, same as timestamps are now**.  (I don't think Jon's screenshot above shows that part.)

**Comment by markerikson** [comment]
Just put up a proof of concept PR over in https://github.com/replayio/devtools/pull/7145 to see how this looks in practice.  Initial running example:

![image](https://user-images.githubusercontent.com/1128784/173150844-83dd46b3-34f9-404b-aa81-89e1f1bf0c46.png)
